### ISSUE TITLE
Sudden Fall of Hashrate in 3060Ti LHR

### ISSUE THREAD

**Issue by sheminasalam** [trexminer/T-Rex#1064]
T rex miner was working without any issues till yesterday and was giving average hash rate of 43.5 with 74% LHR and 140 power limit. Yesterday there was an upgrade for nvidia driver from 470 to 510 and after that there is an issue of sudden fall of hashrate as seen in the graph. I have reverted back to 470 driver but the problem persists. I have used greenwithenvy to overclock with cpu offset of 100 and memory offset of 830. please advise what to do?
![image](https://user-images.githubusercontent.com/45196177/153019998-ebceb686-fd85-4a41-b250-b3badd69cc61.png)

**Comment by trexminer** [comment]
Use 472.12 video driver for best performance. Make sure to completely remove your old driver with DDU.

**Comment by sheminasalam** [comment]
This is running in ubuntu. I think that 472.12 is a windows driver. Do you have any solutions for ubuntu?

**Comment by trexminer** [comment]
Oh I see. 470 series should be fine on ubuntu. Maybe you didn't fully remove the old driver?
Something has definitely changed between your old system (prior to the driver upgrade) and the current one (after the downgrade).

**Comment by sheminasalam** [comment]
I have completely removed nvidia driver and its depenties with the following way.
To remove all the proprietary nvidia packages run the following commands.
```
sudo apt update
sudo apt remove '^nvidia'
sudo apt autoremove 
sudo reboot
```
Then to reinstall the nvidia-driver-470
```
sudo apt update
sudo apt install nvidia-driver-470 
sudo reboot
```
So it should have reverted the nvidia drivers fully.

Any advises?

**Comment by sheminasalam** [comment]
Yes. I sorted it out. My issue was the driver. I completely removed Nvidia proprietary drivers from ubuntu as I have mentioned in the previous post. Then I downloaded [nvidia genuine 470.86](https://www.nvidia.com/Download/driverResults.aspx/183575/en-us) and installed it. 

While installing nvidia original driver you have to first [black list nouvea in ubuntu](https://linuxconfig.org/how-to-disable-blacklist-nouveau-nvidia-driver-on-ubuntu-20-04-focal-fossa-linux)

```
sudo bash -c "echo blacklist nouveau > /etc/modprobe.d/blacklist-nvidia-nouveau.conf"
sudo bash -c "echo options nouveau modeset=0 >> /etc/modprobe.d/blacklist-nvidia-nouveau.conf"
cat /etc/modprobe.d/blacklist-nvidia-nouveau.conf
sudo update-initramfs -u
sudo reboot
```

Then [install linux drivers](https://linuxconfig.org/how-to-install-the-nvidia-drivers-on-ubuntu-20-10-groovy-gorilla-linux). 

```
sudo apt install build-essential libglvnd-dev pkg-config
sudo bash <NVIDIA-driver-run-file-with-location>
sudo reboot
```

Configure NVIDIA X Server Settings.
After reboot your should be able to start NVIDIA X Server Settings app from the Activities menu.
